app/models/activity_insight_oa_status_exporter.rb

```diff
@@ -0,0 +1,55 @@
+# frozen_string_literal: true
+
+class ActivityInsightOAStatusExporter < ActivityInsightExporter
+  def initialize(files)
+    super()
+    @files = files
+  end
+
+  def export
+    logger = Logger.new('log/ai_oa_status_export.log')
+    logger.info "Open access status export to Activity Insight started at #{DateTime.now}"
+    not_exported_ids = []
+    files.each do |file|
+      response = HTTParty.post webservice_url, body: to_xml(file),
+                                               headers: { 'Content-type' => 'text/xml' }, basic_auth: auth, timeout: 180
+      if response.code == 200
+        publication = file.publication
+        publication.exported_oa_status_to_activity_insight = true
+        publication.save!
+      else
+        logger.error Nokogiri::XML(response.to_s).text
+        logger.error "File ID: #{file.id}"
+        not_exported_ids << file.id
+      end
+    end
+
+    logger.info "Open access status export to Activity Insight ended at #{DateTime.now}"
+    logger.info "Files not exported to AI: #{not_exported_ids}"
```

Comment:
Instead of doing all the logging here, we can just raise some kind of error on line 21 if we get an error response from Activity Insight.  Delayed job will then store the error and should also tell us the id of the file that had the error.

Reply:
Also, we'll want to set this: `publication.exported_oa_status_to_activity_insight = true` in the OAWorkflowService before queuing the job.